indico/modules/oauth/models/applications.py

```diff
@@ -0,0 +1,107 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2015 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+from uuid import uuid4
+
+from sqlalchemy.dialects.postgresql import ARRAY
+
+from indico.core.db import db
+from indico.modules.oauth import logger
+from indico.util.string import return_ascii
+
+
+class OAuthApplication(db.Model):
+    """OAuth applications registered in Indico"""
+
+    __tablename__ = 'applications'
+    __table_args__ = {'schema': 'oauth'}
```

Comment:
I'd add an unique index on the lowercase name - see the `LocalGroup` model on how to do it